lib/ansible/plugins/loader.py

```diff
@@ -230,6 +230,7 @@ def __init__(self, class_name, package, config, subdir, aliases=None, required_b
 
         # caches
         self._module_cache = MODULE_CACHE[class_name]
+        self._loaded_module_names = set()
```

Comment:
we already have this data and we should not keep additional cache, if the current config query is not working correctly we just need to fix that.

also this only changes one of many paths to load a plugin, they all need to be consistent.

Reply:
diff cache, I was talking about  ConfigManager, see my PR/diff post for details

Reply:
The consistency is about 'paths taken' to load a plugin, which you have several, the one you modified being only one. by modifying the function as i did, it covers all paths that call it.